src/bio/bio_config.c

```diff
@@ -475,6 +475,71 @@ add_traddrs_from_bdev_subsys(struct json_config_ctx *ctx, bool vmd_enabled,
 	return rc;
 }
 
+#define BDEV_NAME_MAX_LEN	256
+
+static int
+check_name_from_bdev_subsys(struct json_config_ctx *ctx)
+{
+	struct config_entry	 cfg = {};
+	struct spdk_json_val	*key, *value;
+	char			*name;
+	int			 rc = 0;
+	int			 roles = 0;
+
+	D_ASSERT(ctx->config_it != NULL);
+
+	rc = spdk_json_decode_object(ctx->config_it, config_entry_decoders,
+				     SPDK_COUNTOF(config_entry_decoders), &cfg);
+	if (rc < 0) {
+		D_ERROR("Failed to decode config entry: %s\n", strerror(-rc));
+		return -DER_INVAL;
+	}
+
+	if (strcmp(cfg.method, NVME_CONF_ATTACH_CONTROLLER) != 0 &&
+	    strcmp(cfg.method, NVME_CONF_AIO_CREATE) != 0) {
```

Comment:
Is it for supporting AIO device, right? It looks to me control plane now can't generate proper roles for AIO device, maybe we'd ask Tom to support it, AIO device is quite useful for development & test.

Reply:
I tested AIO device locally, it works for me.